README.md

```diff
@@ -1,24 +1,50 @@
-Nginx SSL Self-Signed
-=====================
+SSL Certificates
+================
+
+Manage SSL certificates for web-servers.
+
+Optionally generate self-signed SSL certificates for internal testing.
 
-Generate self-signed SSL certificates for Nginx, for internal testing.
 
 Role Variables
 --------------
 
 Defaults: `defaults/main.yml`
 
 Optional variables:
+- `ssl_certificate_path`: Server path to SSL certificate
+- `ssl_certificate_key_path`: Server path to SSL certificate key
+- `ssl_certificate_combined_path`: Server path to SSL combined certificate and key (e.g. for Haproxy), set to empty to disable
+- `ssl_certificate_content`: Text content of the certificate, for instance from vault
+- `ssl_certificate_key_content`: Text content of the certificate key
+- `ssl_certificate_selfsigned_create`: Create a self-signed certificate if necessary, default `True`
```

Comment:
From  a consumer perspective this boolean variable is self-describing. In terms of the logic of the role, it reads fully equivalent to `ssl_certificate_content != '' && ssl_key_content != ''`. Is there a reason for using a separate boolean variable? What should be the behavior is `ssl_certificate_selfsigned_create` is `False` but only one of the two `ssl_xxx_content` variables is set (should this fail an early check) ?

Reply:
I thought it was worth being explicit, so that you can use this role and ensure it never creates a self-signed cert. I've added a check to ensure the 3 variables are consistent.